src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -60,36 +73,76 @@ private void OnSolutionCrawlerProgressChanged(object sender, ProgressData progre
                 case ProgressStatus.Started:
                     StartedOrStopped(started: true);
                     break;
-                case ProgressStatus.Updated:
-                    ProgressUpdated(progressData.FilePathOpt);
+                case ProgressStatus.PendingItemUpdated:
+                    _lastPendingItemCount = progressData.PendingItemCount.Value;
+                    ProgressUpdated();
                     break;
                 case ProgressStatus.Stoped:
                     StartedOrStopped(started: false);
                     break;
+                case ProgressStatus.Evaluating:
+                case ProgressStatus.Paused:
+                    _lastProgressStatus = progressData.Status;
+                    ProgressUpdated();
+                    break;
                 default:
                     throw ExceptionUtilities.UnexpectedValue(progressData.Status);
             }
         }
 
-        private void ProgressUpdated(string filePathOpt)
+        private void ProgressUpdated()
         {
+            // we prefer showing evaluating if progress is flipping between evaluate and pause
+            // in short period of time.
+            var forceUpdate = _lastShownProgressStatus == ProgressStatus.Paused &&
+                              _lastProgressStatus == ProgressStatus.Evaluating;
+
             var current = DateTimeOffset.UtcNow;
-            if (current - _lastTimeReported < s_minimumInterval)
+            if (!forceUpdate && current - _lastTimeReported < s_minimumInterval)
             {
                 // make sure we are not flooding UI. 
-                // this is just presentation, fine to not updating UI especially since
+                // this is just presentation, fine to not updating UI right away especially since
                 // at the end, this notification will go away automatically
+                // but to make UI to be updated to right status eventually if task takes long time to finish
+                // we enqueue refresh task.
+                EnqueueRefresh();
                 return;
             }
 
+            _lastShownProgressStatus = _lastProgressStatus;
             _lastTimeReported = current;
-            ChangeProgress(_taskHandler, filePathOpt != null ? string.Format(ServicesVSResources.Analyzing_0, FileNameUtilities.GetFileName(filePathOpt)) : null);
+
+            ChangeProgress(_taskHandler, GetMessage());
+
+            string GetMessage()
+            {
+                var status = (_lastProgressStatus == ProgressStatus.Paused) ? ServicesVSResources.Paused : ServicesVSResources.Evaluating;
+                return $"{status} ({_lastPendingItemCount} {ServicesVSResources.tasks_in_queue})";
```

Comment:
❕ This is not friendly to localization (it makes assumptions about sentence structure and punctuation).

Requesting @tmeschter for guidance on the current recommended form for this resource string.

Reply:
@sharwell is correct: this string cannot be properly localized as-is. The major problem here is that you can't take individually-localized strings and combine them like this. Doing so makes assumptions about word order, spacing, and punctuation that don't hold for all languages, and the final, "translated" string will be meaningless.

Do this instead:

``` C#
var formatString = (_lastProgressStatus == ProgressStatus.Paused)
    ? ServicesVSResources.DiagnosticProgressReporterPausedMessage
    : ServicesVSResources.DiagnosticProgressReporterEvaluatingMessage;
return string.Format(formatString, _lastPendingItemCount);
```

Where `DiagnosticProgressReporterPausedMessage` is something like `Paused ({0} tasks in queue)`, and `DiagnosticProgressReporterEvaluatingMessage` is `Evaluating ({0} tasks in queue)`.

Reply:
Oh, and in the .resx file please document what the placeholders (e.g. {0}) represent in the notes/comments field.

Reply:
sure.